### ISSUE TITLE
Incorrect warning for `Found a configuration for 'some_rule' rule, but it is not enabled on 'opt_in_rules'.` with `parent_config`'s

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#4858]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

All examples run from the top of the SwiftLint source tree.

I made a new configuration file:

```
% cat .swiftlint-new.yml 
included:
  - Plugins
  - Source
  - Tests
excluded:
  - Tests/SwiftLintFrameworkTests/Resources
opt_in_rules:
  - implicit_return
disabled_rules:
  - type_body_length
  - function_body_length
  - large_tuple

implicit_return:
  included:
    - closure
#   - function
#   - getter
```

Running it from the top of the tree:

```
% swiftlint --config .swiftlint-new.yml  --reporter summary --quiet
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| rule identifier | opt-in | correctable | custom | warnings | errors | total violations | number of files |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| implicit_return | yes    | yes         | no     |       59 |      0 |               59 |              36 |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| Total           |        |             |        |       59 |      0 |               59 |              36 |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
```

Now I add a new configuration, whose parent is the one I just created, but which is stricter about the settings for `implicit_return`:

```
% cat .swiftlint-child.yml
parent_config: .swiftlint-new.yml

implicit_return:
  included:
    - closure
    - function
    - getter
```

Running SwiftLint against this config:

```
warning: Found a configuration for 'implicit_return' rule, but it is not enabled on 'opt_in_rules'.
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| rule identifier | opt-in | correctable | custom | warnings | errors | total violations | number of files |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| implicit_return | yes    | yes         | no     |      448 |      0 |              448 |             200 |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
| Total           |        |             |        |      448 |      0 |              448 |             200 |
+-----------------+--------+-------------+--------+----------+--------+------------------+-----------------+
```

`rules` also produces the warning, but indicates that the new configuration is applied:

```
% swiftlint  rules --config .swiftlint-child.yml                            
warning: Found a configuration for 'implicit_return' rule, but it is not enabled on 'opt_in_rules'.
+------------------------------------------+--------+-------------+------------------------+-------------+----------+----------------+---------------------------------------------------------------------------------------+
| identifier                               | opt-in | correctable | enabled in your config | kind        | analyzer | uses sourcekit | configuration                                                                         |
+------------------------------------------+--------+-------------+------------------------+-------------+----------+----------------+---------------------------------------------------------------------------------------+
| implicit_return                          | yes    | yes         | yes                    | style       | no       | yes            | severity: warning, included: [closure, function, getter]                              |
+------------------------------------------+--------+-------------+------------------------+-------------+----------+----------------+---------------------------------------------------------------------------------------+
```

Expected: if the rule is enabled, even in another config, the warning should not be displayed.

##### Complete output when running SwiftLint, including the stack trace and command used

See above:

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.51.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

See above

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No, but parent_config yes - see description above

* Which Xcode version are you using (check `xcodebuild -version`)?

```
Xcode 14.2
Build version 14C18
```

* Do you have a sample that shows the issue? 

See description above.

**Comment by tinder-garricnahapetian** [comment]
I'm seeing this as well. Main difference is that we are using `only_rules`, not `opt_in_rules`.

`Found a configuration for 'empty_count' rule, but it is not enabled on 'opt_in_rules'.`

Configs:

```
% cat .swiftlint.yml
included:
- Sources

child_config: .swiftlint-rules.yml

empty_count:
  only_after_dot: true
```

```
% cat .swiftlint-rules.yml
only_rules:
- empty_count
```

**Comment by mildm8nnered** [comment]
I have a draft PR that fixes this - I'll take a look at that case as well

**Comment by mildm8nnered** [comment]
@tinder-garricnahapetian - so your case above is related, but is actually slightly different in that you're using `child_config`.

I'm going to open a separate issue for that
